ui.apps/src/main/content/jcr_root/apps/etoolbox-rollout-manager/clientlibs/rollout-manager-ui/js/console-ui.actions.js

```diff
@@ -25,16 +25,13 @@
      * Retrieves data related to eligible for synchronization live copies as a json array. The data is
      * used for building 'Targets' tree in the UI dialog
      * @param path - path of the page selected in Sites
-     * @returns {*}
+     * @returns {Promise}
      */
     function collectLiveCopies(path) {
         return $.ajax({
             url: COLLECT_LIVE_COPIES_COMMAND,
             type: 'POST',
-            data: {
-                _charset_: 'UTF-8',
-                path
-            }
+            data: { _charset_: 'UTF-8', path }
```

Comment:
Do we need to add timeout option to AJAX requests to handle slow or unresponsive servers gracefully?

Reply:
good question, we can add that)
do you think **30 sec** for _Data collection/Blueprint check_ and **60 sec** for _Rollout_ will be enough?

Reply:
It depends on web server and browser configurations. Browsers usually have 300 seconds. But web servers have different default values for request timeout. Usually 60-300. It needs to be checked.

Reply:
For aem as a cloud: 
"For code that does not apply these timeouts, AEM instances running on AEM as a Cloud Service will enforce a global timeouts. These timeout values are 10 seconds for connect calls and 60 seconds for read calls for connections."